File: library-frontend/library-front/src/app/services/book.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {Book} from '../models/book';
import {Api} from '../../util/api';
import {Opinion} from '../models/opinion';

@Injectable({
  providedIn: 'root'
})
export class BookService {

  private book: Book;

  constructor(
    private http: HttpClient) {
  }

  setBookContext(book: Book) {
    this.book = book;
  }

  getBookContext() {
    return this.book;
  }

  public save(book: Book): Observable<any> {
    return this.http.post(`${Api.BOOKS_END_POINT}/add`, book);
  }

  getBooksOrderedByAvgRate(): Observable<Book[]> {
    return this.http.get<Book[]>(`${Api.BOOKS_END_POINT}/ratings`);
  }

  getClassicLibraryBooks(): Observable<Book[]> {
    return this.http.get<Book[]>(`${Api.BOOKS_END_POINT}/classic-library`);
  }

  getUserRentalServiceBooks(): Observable<Book[]> {
    return this.http.get<Book[]>(`${Api.BOOKS_END_POINT}/rental-service`);
  }

  deleteBookById(bookId: number): Observable<any> {
    return this.http.delete(`${Api.BOOKS_END_POINT}/delete/` + bookId);
  }

  addOpinion(opinion: Opinion) {
    return this.http.post(`${Api.BOOKS_END_POINT}/add-opinion/`, opinion);

  }

  findBookById(bookId: number): Observable<Book> {
    return this.http.get<Book>(`${Api.BOOKS_END_POINT}/` + bookId);
  }

}

File: library-spring/spring-library/books/src/main/java/pp/books/service/BookService.java
package pp.books.service;


import pp.books.domain.Book;
import pp.books.dto.BookUpdateStatusDto;
import pp.books.dto.OpinionDto;

import java.util.List;

public interface BookService {

    void addBook(Book book);

    List<Book> getClassicLibraryBooks();

    List<Book> getUserRentalServiceBooks();

    void deleteBookById(long bookId);

    Book getBookById(long bookId);

    void updateBook(Book book);

    void addOpinion(OpinionDto opinion);

    List<Book> getBooksOrderedByAvgRate();

    void updateBookStatus(BookUpdateStatusDto bookUpdateStatusDto);
}
